src/js/plugins/bluetooth.js

```diff
@@ -116,53 +209,85 @@ class Bluetooth extends Plugin {
     /* ----------------------------- BUTTONS ------------------------------*/
 
     scanForDevices() {
-        var idx = this.devicesListEl.selectedIndex;
-        api.putPlugin(this.callsign, 'StartScan', null, (err, resp) => {
+        this.status(`Start scanning`);
+        api.putPlugin(this.callsign, 'Scan', null, (err, resp) => {
             if (err !== null) {
                 console.error(err);
                 return;
             }
 
-            // get the results
-            // FIXME, there should be a seperate API to provide results on the "scanned results". For now go back to this.update
-            setTimeout(this.update.bind(this), 5000);
+            // update after 2s
+            setTimeout(this.update.bind(this), 2000);
+
+            // update discovered device list in every 4s
+            this.Timer = setInterval(this.getDiscoveredDevices.bind(this), 4000);
+
+            this.status(`Scanning...`);
+            // stop scan after 1 min
+            setTimeout(this.stopScan.bind(this), 60000);
+
         });
     }
 
-    pairDevice() {
-        var idx = this.devicesListEl.selectedIndex;
+    stopScan() {
+        this.status(`Stopping Scan`);
+        api.putPlugin(this.callsign, 'StopScan', null, (err, resp) => {
+            if (err !== null) {
+                console.error(err);
+                return;
+            }
 
-        this.status(`Pairing with ${this.devices[idx].name}`);
+            clearInterval(this.Timer);
+            setTimeout(this.update.bind(this), 2000);
+            this.status(`Scan stopped`);
+        });
+    }
+
+    pairDevice() {
+        var val = JSON.parse(document.getElementById('BT_DiscoveredDevices').value);
+        if (val.Name === "")
+            this.status(`Pairing with ${val.Address}`);
+        else
+            this.status(`Pairing with ${val.Name}`);
 
-        api.putPlugin(this.callsign, 'PairDevice', this,devices[idx].id, (err, resp) => {
+        api.putPlugin(this.callsign, 'Pair', '{"Address" : "' + val.Address + '"}', (err, resp) => {
             if (err !== null) {
                 console.error(err);
                 return;
             }
 
-            // get the results
-            // FIXME, there should be a seperate API to provide results on the "scanned results"
-            setTimeout(this.update.bind(this), 5000);
+            // update Paired device list after 2s
+            setTimeout(this.getPairedDevices.bind(this), 2000);
+            setTimeout(this.renderPairedDevices.bind(this), 3000);
```

Comment:
We already call the renderedPairedDevices as a result of succesfully retrieving the data in getPairedDevices, I dont think the second setTimeout is required here.